src/inverseRobot.js

```diff
@@ -7,7 +7,25 @@
 */
 
 function inverseRobot(robot) {
-  // write code here
+  const values = Object.values(robot);
+
+  const uniqueValues = new Set(values);
+
+  if (uniqueValues.size !== values.length) {
+    return null;
+  }
+
+  const invertedRobot = { };
+
+  for (const key in robot) {
+    const value = robot[key];
+
+    if (!Object.prototype.hasOwnProperty.call(invertedRobot, key)) {
+      invertedRobot[value] = key;
+    }
```

Comment:
you don't have to validate this condition, it's handled by the condition above
btw, `hasOwnProperty` call could've been simplified, no need to access via prototype
```
const object = {...};

if (object.hasOwnProperty(...)) {
...
```